cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -573,41 +578,33 @@ func (s *Server) getClient(ctx context.Context, namespace string) (ctrlclient.Cl
 	return ctrlclient.NewNamespacedClient(client, namespace), nil
 }
 
-// GetPluginDetail returns a core.plugins.Plugin describing itself.
-func GetPluginDetail() *plugins.Plugin {
-	return common.GetPluginDetail()
-}
-
-// parsePluginConfig parses the input plugin configuration json file and return the
-// configuration options.
-func parsePluginConfig(pluginConfigPath string) (pkgutils.VersionsInSummary, int32, error) {
-	// Note at present VersionsInSummary is the only configurable option for this plugin,
-	// and if required this func can be enhanced to return fluxConfig struct
-
-	// In the flux plugin, for example, we are interested in config for the
-	// core.packages.v1alpha1 only. So the plugin defines the following struct and parses the config.
-	type fluxPluginConfig struct {
-		Core struct {
-			Packages struct {
-				V1alpha1 struct {
-					VersionsInSummary pkgutils.VersionsInSummary
-					TimeoutSeconds    int32 `json:"timeoutSeconds"`
-				} `json:"v1alpha1"`
-			} `json:"packages"`
-		} `json:"core"`
-	}
-	var config fluxPluginConfig
-
-	pluginConfig, err := ioutil.ReadFile(pluginConfigPath)
+// hasAccessToNamespace returns an error if the client does not have read access to a given namespace
+func (s *Server) hasAccessToNamespace(ctx context.Context, namespace string) (bool, error) {
+	typedCli, err := s.clientGetter.Typed(ctx, s.kubeappsCluster)
 	if err != nil {
-		return pkgutils.VersionsInSummary{}, 0, fmt.Errorf("unable to open plugin config at %q: %w", pluginConfigPath, err)
-	}
-	err = json.Unmarshal([]byte(pluginConfig), &config)
+		return false, err
+	}
+
+	res, err := typedCli.AuthorizationV1().SelfSubjectAccessReviews().Create(
+		context.TODO(),
```

Comment:
Why not the original `ctx`? (it may have deadlines or cancellations etc. associated with it).

Reply:
it was a copy and paste of code from helm plugin. I will change it in mine. I assume you'll take care of helm

Reply:
I don't see those boundaries the way you do - I don't mind if you update it in helm too (in fact, it'd seem normal). But no problem if you don't want to.